modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/infrastructure/snapshot/GlobalSnapshotConsensusStateCreator.scala

```diff
@@ -31,39 +36,95 @@ object GlobalSnapshotConsensusStateCreator {
+    clusterStorage: ClusterStorage[F]
   ): GlobalSnapshotConsensusStateCreator[F] = new GlobalSnapshotConsensusStateCreator[F] {
+    val logger: SelfAwareStructuredLogger[F] = Slf4jLogger.getLoggerFromName[F](this.getClass.getName)
+
     def tryFacilitateConsensus(
       key: GlobalSnapshotKey,
       lastOutcome: GlobalConsensusOutcome,
       maybeTrigger: Option[ConsensusTrigger],
-      resources: ConsensusResources[GlobalSnapshotArtifact, GlobalConsensusKind]
+      resources: ConsensusResources[GlobalSnapshotArtifact, GlobalConsensusKind],
+      firstConsensusRound: Boolean = false
     ): F[StateCreateResult] =
       consensusStorage
-        .condModifyState(key)(toCreateStateFn(facilitateConsensus(key, lastOutcome, maybeTrigger, resources)))
+        .condModifyState(key)(toCreateStateFn(facilitateConsensus(key, lastOutcome, maybeTrigger, resources, firstConsensusRound)))
         .flatMap(evalEffect)
         .flatTap(logIfCreatedState)
 
     private def facilitateConsensus(
       key: GlobalSnapshotKey,
       lastOutcome: GlobalConsensusOutcome,
       maybeTrigger: Option[ConsensusTrigger],
-      resources: ConsensusResources[GlobalSnapshotArtifact, GlobalConsensusKind]
+      resources: ConsensusResources[GlobalSnapshotArtifact, GlobalConsensusKind],
+      firstConsensusRound: Boolean = false
     ): F[(GlobalSnapshotConsensusState, F[Unit])] =
       for {
 
         candidates <- consensusStorage.getCandidates(key.next)
 
-        facilitators <- lastOutcome.facilitators.value
-          .concat(lastOutcome.finished.candidates.value)
+        // Split into old facilitators and new candidates
+        oldFacilitators = lastOutcome.facilitators.value
           .filter(peerId => seedlist.forall(_.map(_.peerId).contains(peerId)))
-          .filterA(consensusFns.facilitatorFilter(lastOutcome.finished.signedMajorityArtifact, lastOutcome.finished.context, _))
-          .map(_.prepended(selfId).distinct.sorted)
+
+        newCandidates = lastOutcome.finished.candidates.value
+          .filter(peerId => seedlist.forall(_.map(_.peerId).contains(peerId)))
+
+        // Check responsiveness for each old facilitator
+        oldFacilitatorsWithStatus <- oldFacilitators.traverse { peerId =>
+          if (peerId === selfId) {
+            Applicative[F].pure((peerId, true, "self".some))
+          } else {
+            clusterStorage.getPeer(peerId).map {
+              case Some(peer) if peer.responsiveness === Responsive =>
+                (peerId, true, "responsive".some)
+              case Some(peer) if peer.responsiveness === Unresponsive =>
+                (peerId, false, "unresponsive".some)
+              case Some(peer) =>
+                (peerId, false, peer.responsiveness.show.some)
+              case None =>
+                (peerId, false, "not found in cluster storage".some)
+            }
+          }
+        }
+
+        removedFacilitators = oldFacilitatorsWithStatus.filterNot(_._2)
```

Comment:
Tuple here leads to _._2 which is hard to read, especially if moved in different areas.